### PR TITLE
Update the mobile snippet preview.

### LINKED ISSUE #300: Change mobile snippet preview to reflect Google's changes
Roadmap issue: https://github.com/Yoast/roadmap/issues/101
Product owner: @jdevalk 

## Is your feature request related to a problem? Please describe.
Google has tweaked how it displays mobile snippets. Our plugin's styling is therefore outdated.

## Describe the solution you'd like
- 12px padding top and bottom.
- 16px padding left and right.
- The URL is moved above the title, font size 12px with color #3C4043.
- A 16x16 favicon is added in front of the URL. Use the default globe when no favicon is passed to the component.
- SEO title color is #4B11A8.
- Meta description text is 14px with 20px line-height and color #3C4043. Its max length is 150 characters, whether there is an image or not.
- image is 104x104px. Don't show a placeholder when no image is passed to the component.
- 8px border radius for paper and image.
- box shadow 0 1px 6px rgba(32, 33, 36, 0.28).
- max-width 400px.

Screen Shot 2019-05-27 at 13 07 20

Screen Shot 2019-05-27 at 13 07 20

The globe:

```
data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABAAAAAQCAYAAAAf8/9hAAABs0lEQVR4AWL4//8/RRjO8Iucx+noO0MWUDo16FYABMGP6ZfUcRnWtm27jVPbtm3bttuH2t3eFPcY9pLz7NxiLjCyVd87pKnHyqXyxtCs8APd0rnyxiu4qSeA3QEDrAwBDrT1s1Rc/OrjLZwqVmOSu6+Lamcpp2KKMA9PH1BYXMe1mUP5qotvXTywsOEEYHXxrY+3cqk6TMkYpNr2FeoY3KIr0RPtn9wQ2unlA+GMkRw6+9TFw4YTwDUzx/JVvARj9KaedXRO8P5B1Du2S32smzqUrcKGEyA+uAgQjKX7zf0boWHGfn71jIKj2689gxp7OAGShNcBUmLMPVjZuiKcA2vuWHHDCQxMCz629kXAIU4ApY15QwggAFbfOP9DhgBJ+nWVJ1AZAfICAj1pAlY6hCADZnveQf7bQIwzVONGJonhLIlS9gr5mFg44Xd+4S3XHoGNPdJl1INIwKyEgHckEhgTe1bGiFY9GSFBYUwLh1IkiJUbY407E7syBSFxKTszEoiE/YdrgCEayDmtaJwCI9uu8TKMuZSVfSa4BpGgzvomBR/INhLGzrqDotp01ZR8pn/1L0JN9d9XNyx0AAAAAElFTkSuQmCC
```

## Important
Alter the component in such a way that it doesn't break anything when it's used in the plugin with the current props/data. Make sure the default favicon is in place, and the component can handle no image being passed.

### PR DESCRIPTION
**Pull request by afercia**
## Summary

~~Work In Progress, pushing for some feedback. Please don't merge~~ 🙂 
[Edit]: questions have been answered, PR ready to be reviewed.

Questions:
- seems to me `#4b11a8` in the issue instruction is actually the color for _visited_ link while the one for normal links is `#1967d2`
- seems to me the description max length is already implemented (150 characters) 
- additional changes:
  - I've updated the SEO title font size to 16px (was 18)
  - updated the Date color 
  - removed the separator 

Screenshot without image and with default globe favicon:

Screenshot 2019-05-31 at 10 19 59

Screenshot with image and with custom globe favicon:

Screenshot 2019-05-31 at 10 21 06

  * Package(s) involved: search-metadata-previews

  * Should the change be included in the package changelog?
    * [ ] No
    * [x] Yes
  * Should the change be included in one or more plugin changelogs?
    * [ ] No
    * [X] Free
    * [X] Premium
    * [ ] Other (please specify)
  * Package changelog item (if applicable): Updated the Google Mobile Snippet Preview
  * Plugin changelog item (if applicable): Updated the Google Mobile Snippet Preview

## Test instructions

This PR can be tested by following these steps:

- pass a custom favicon e.g.: `data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABAAAAAQCAMAAAAoLQ9TAAAAmVBMVEVHcEyaHVSdA17httKkKGqkKGqkKGqkKGqkKGqkJ2qkKGqkKGr5/PXw/OXb68j///////3Snbn////u1uT+//2kKGpnrQDZ6cWhD2SjIWefGmRdqACfA2Hg28+GAEyey2rQ6rl1tRrfzM6LwUrcuc/Rp8B8uSrn6tmRAlSjV4G43ZbJg6r0/+3G5qymzniyhJ6JJFyvO33B4KXlR4xrAAAAFXRSTlMALYeqRMHHxoqNas3Oy+2Bxehf36eZkf+lAAAAo0lEQVQYlVXP1w6DMAwFUDbd29khYdMCXf//cU2Aova+2D5SFNtx5lx0fXJ+gzkeG3dp4/kqPbt2DoASwkBk6HalgYGQtp0uCWgkEwgNRCAwl0mZ5yWDyEAMpKh6JpF+AsQD0FdfFW+UNV8AIXlXqQebgbVKIZzACJGpAqO0aCYILQxLTBCYJ8R+bGZYD6t7xzq9+97CZDVes+X54e/OzW4/9x/C9xAwpoxo+QAAAABJRU5ErkJggg==`
- pass an image to be displayed in the description e.g.: `https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcTUmFOm_iWPqNYy_fws7zl_66zItEuHLPAIuz5qX3Ka5wRr5tSzcTPRXfmElood_YTLlIQkEw&usqp=CAE`
- check the Mobile preview looks as intended
- check there are no visual changes in the Desktop preview
- check the "caret" icons displayed on the left when hovering the snippet parts still work correctly

## Impact check

* This PR affects the following parts of the plugin, which may require extra testing:
  * Meta box snippet preview (mobile)
  * check everything works including the "caret" icons on the left

## UI changes
* [x] This PR changes the UI in the plugin. I have added the 'UI change' label to this PR.

Fixes #300

### CODE REVIEW

**Comment by afercia** [comment]
Aside: noticed we're passing a `width` prop that ends up in a DOM element: this is invalid HTML and should be fixed. Will have a look at it.

Screenshot 2019-05-31 at 10 10 44

**Comment by afercia** [comment]
Last commit fixes the invalid `width` HTML attribute rendered on DOM nodes.

**Comment by hedgefield** [comment]
About the colors, you suppose right ;) good catch. Looks good from those screenshots!

**Comment by afercia** [comment]
Screenshot with the new mobile preview in the plugin meta box:

Screenshot 2019-06-03 at 09 31 49

File: packages/search-metadata-previews/src/snippet-preview/SnippetPreview.js [hunk 1]
```diff
@@ -655,19 +728,24 @@ export default class SnippetPreview extends PureComponent {
...
 							{ __( "SEO title preview", "yoast-components" ) + ":" }
 						</ScreenReaderText>
+						{ ! isDesktopMode && this.renderUrl() }
 						<SnippetTitle
 							onMouseUp={ onMouseUp.bind( null, "title" ) }
 							onMouseEnter={ onMouseEnter.bind( null, "title" ) }
 							onMouseLeave={ onMouseLeave.bind( null ) }
 						>
-							<TitleBounded>
+							<TitleBounded screenMode={ mode }>
```
Comment (igorschoester) [inline packages/search-metadata-previews/src/snippet-preview/SnippetPreview.js]: Correct me if I'm wrong, but I don't see `screenMode` getting used for this component or below.

File: packages/search-metadata-previews/src/snippet-preview/SnippetPreview.js [hunk 2]
```diff
@@ -170,10 +214,23 @@ const DatePreview = styled.span`
...
+	"yqXyxtCs8APd0rnyxiu4qSeA3QEDrAwBDrT1s1Rc/OrjLZwqVmOSu6+Lamcpp2KKMA9PH1BY" +
+	"XMe1mUP5qotvXTywsOEEYHXxrY+3cqk6TMkYpNr2FeoY3KIr0RPtn9wQ2unlA+GMkRw6+9TF" +
+	"w4YTwDUzx/JVvARj9KaedXRO8P5B1Du2S32smzqUrcKGEyA+uAgQjKX7zf0boWHGfn71jIKj" +
+	"2689gxp7OAGShNcBUmLMPVjZuiKcA2vuWHHDCQxMCz629kXAIU4ApY15QwggAFbfOP9DhgBJ" +
+	"+nWVJ1AZAfICAj1pAlY6hCADZnveQf7bQIwzVONGJonhLIlS9gr5mFg44Xd+4S3XHoGNPdJl" +
+	"1INIwKyEgHckEhgTe1bGiFY9GSFBYUwLh1IkiJUbY407E7syBSFxKTszEoiE/YdrgCEayDmt" +
+	"aJwCI9uu8TKMuZSVfSa4BpGgzvomBR/INhLGzrqDotp01ZR8pn/1L0JN9d9XNyx0AAAAAElF" +
+	"TkSuQmCC";
+
+const FavIcon = styled.img`
```
Comment (igorschoester) [inline packages/search-metadata-previews/src/snippet-preview/SnippetPreview.js]: Same reasoning as above.

```suggestion
const Favicon = styled.img`
```

File: packages/search-metadata-previews/src/snippet-preview/SnippetPreview.js [hunk 3]
```diff
@@ -132,25 +137,64 @@ const BaseUrlOverflowContainer = styled( BaseUrl )`
...
+	display: block;
+	width: 104px;
+	height: 104px;
 `;
 
 const MobilePartContainer = styled.div`
-	padding: 8px 16px;
+	padding: 12px 16px;
+
+	&:not(:last-child) {
```
Comment (igorschoester) [inline packages/search-metadata-previews/src/snippet-preview/SnippetPreview.js]: This rule seems strange to me. Could you enlighten me a little to what this solves and why you chose to do it this way?

File: packages/search-metadata-previews/src/snippet-preview/SnippetPreview.js [hunk 4]
```diff
@@ -412,7 +469,7 @@ export default class SnippetPreview extends PureComponent {
 	renderDate() {
 		return this.props.date === ""
 			? null
-			: <DatePreview>{ this.props.date } - </DatePreview>;
+			: <DatePreview>{ this.props.date } · </DatePreview>;
```
[the remark this answers is not in the document]
Comment (afercia) [reply packages/search-metadata-previews/src/snippet-preview/SnippetPreview.js]: Yeah, good catch!

File: packages/search-metadata-previews/src/snippet-preview/SnippetPreview.js (hunk 1 above)
Comment (afercia) [reply packages/search-metadata-previews/src/snippet-preview/SnippetPreview.js]: This prop is passed down from the styled `TitleBounded` to the the styled `Title` and it's needed to differentiate the title color between desktop and mobile.

File: packages/search-metadata-previews/src/snippet-preview/SnippetPreview.js (hunk 3 above)
Comment (afercia) [reply packages/search-metadata-previews/src/snippet-preview/SnippetPreview.js]: `MobilePartContainer` has a default `padding: 12px 16px;` which makes the space between title and description too large. 

with:

Screenshot 2019-06-05 at 09 54 58

without:

Screenshot 2019-06-05 at 09 55 08

The rule is copied from Google 😛 Using `:not()` it's a bit more future-proof in the case more `MobilePartContainer` will be added. Could be simplified to first-child > padding-bottom: 0 at the condition no other `MobilePartContainer` are ever added.

**Comment by afercia** [comment]
Just noticed the "dot" separator in this PR is actually squared:

Screenshot 2019-06-05 at 10 44 22

while the real one on Google is rounded:

Screenshot 2019-06-05 at 10 28 26

going to fit this too, found a unicode character that seems suitable.

**Comment by IreneStr** [comment]
CR 👍

**Comment by hwinne** [comment]
Acceptance 👍

**Pull request merged**